src/Port.php

```diff
@@ -28,8 +28,7 @@ class Port extends AbstractComponent implements Component
      */
     protected function validate($data)
     {
-        $data = filter_var($data, FILTER_VALIDATE_INT, ['options' => ['min_range' => 1]]);
-        if (! $data) {
+        if (! ctype_digit($data) || $data < 1) {
```

Comment:
I'm honestly not sure if this is better. It may be more brittle because `ctype_*` methods trigger warnings when the value is not a string. While the added type casting ensures it won't error, that might still be a good reason to revert this.